lib/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+let R = require('ramda');
+let database = require('./database');
+
+let queries = {
+  measure: (query) => {
+    return `SELECT MEAN(value) as value
+      FROM "measure"
+      WHERE "metric" = '${query.metric}'
+      AND "context" = '${query.context}'
+      AND "branch" = '${query.branch}'
+      AND "device" = '${query.device}'
+      AND "memory" = '${query.memory}'
+      AND "test" = '${query.test}'
+      AND ${query.timeFilter}
+      GROUP BY * fill(none)`;
+  },
+  memory: (query) => {
+    return `SELECT MEAN(value) as value
+      FROM "memory"
+      WHERE "metric" = '${query.metric}'
+      AND "context" = '${query.context}'
+      AND "branch" = '${query.branch}'
+      AND "device" = '${query.device}'
+      AND "memory" = '${query.memory}'
+      AND "test" = '${query.test}'
+      AND ${query.timeFilter}
+      GROUP BY * fill(none)`;
+  },
+  mtbf: (query) => {
+    return `SELECT SUM(value) / SUM(failures) as value
+      FROM "mtbf"
+      WHERE "node" = '${query.node}'
+      AND "branch" = '${query.branch}'
+      AND "device" = '${query.device}'
+      AND "memory" = '${query.memory}'
+      AND ${query.timeFilter}
+      GROUP BY time(1s) fill(none)`;
+  },
+  power: (query) => {
+    return `SELECT value
+      FROM "power"
+      WHERE "test" = '${query.test}'
+      AND "context" = '${query.context}'
+      AND "branch" = '${query.branch}'
+      AND "device" = '${query.device}'
+      AND "memory" = '${query.memory}'
+      AND ${query.timeFilter}`;
+  }
+};
+
+/**
+ * Remove extra whitespace and newlines from a query
+ * @param {String} queryString
+ * @returns {String}
+ */
+let clean = R.pipe(R.replace(/\n[ ]+/g, ' '), R.trim);
+
+/**
+ * Fetch the query string for a given measurement using options to fill in query values
+ * @param {String} measurement InfluxDB measurement
+ * @param {Object} options query values
+ */
+let getRawQuery = (measurement, options) => R.call(R.prop(measurement, queries), options);
+
+/**
+ * Fetch the query string for a given set of options
+ * @param {Object} options
+ * @returns {String}
+ */
+let getQuery = R.converge(R.pipe(getRawQuery, clean), [R.prop('measurement'), R.identity]);
+
+/**
+ * Execute a query against an InfluxDB database
+ * 1. Create a database client
+ * 2. Fetch the query string
+ * 3. Execute the query against the database
+ * @param options
+ * @returns {Promise}
+ */
+module.exports = (options) => {
+  return Promise
+    .all([
+      database(options),
+      getQuery(options)
+    ])
+    .then(R.apply((client, query) => {
+      if (query.includes('undefined')) {
+        throw new Error(`Missing parameter, query contains undefined value:\n\n${query}\n`)
```

Comment:
Yeah the more I think about it this is probably sufficient when there's no --device, like you noted it's required everytime anyway so odds are the end user would figure out that they're missing --device, so i'm fine either way